frontend-react/src/components/Table/Table.tsx

```diff
@@ -90,14 +91,242 @@ export interface TableProps {
+        return tableData(
+            <NavLink
+                className="usa-link"
+                to={`${feature.linkBasePath || ""}${
+                    rowData[feature.linkAttr || field]
+                }`}
+            >
+                {columnConfig.valueMap
+                    ? showMappedFieldValue(columnConfig, rowData)
+                    : displayValue}
+            </NavLink>
+        );
+    }
+
+    if (hasFeature("action")) {
+        // Make column value actionable
+        const { action, param } = columnConfig.feature as ActionableColumn;
+
+        if (!rowData[param!!]) {
+            console.warn(`The row attribute '${param}' could not be found`);
+        }
+
+        const doAction = () => {
+            if (param) return action(rowData[param]);
+            return action();
+        };
+        return tableData(
+            <button
+                className="usa-link bg-transparent border-transparent"
+                onClick={() => doAction()}
+            >
+                {displayValue}
+            </button>
+        );
+    }
+
+    if (isEditing()) {
+        // Make column value editable
+        return tableData(
+            <input
+                aria-label={`${field}-${rowIndex}`}
+                className="usa-input"
+                onChange={(event) => {
+                    setUpdatedRow(event.target.value, field);
+                }}
+                /* This ensures the value seen in the edit field is
+                 * the same as the server-provided data, NOT the
+                 * displayed data (in case of transformation/map) */
+                defaultValue={displayValue}
+            />
+        );
+    }
+
+    return columnConfig.valueMap
+        ? tableData(showMappedFieldValue(columnConfig, rowData))
+        : tableData(displayValue);
+};
+
+interface TableRowProps {
+    rows: TableRow[];
+    columns: ColumnConfig[];
+    filterManager?: FilterManager;
+    enableEditableRows?: boolean;
+    onSave?: (row: TableRow | null) => Promise<void>;
+}
+
+/* Iterates each row, and then uses the key value from columns.keys()
+ * to render each cell in the appropriate column. */
+export const TableRows = ({
+    rows,
+    onSave = () => Promise.resolve(),
+    enableEditableRows,
+    filterManager,
+    columns,
+}: TableRowProps) => {
+    // tracks which row is currently being edited
+    const [editing, setEditing] = useState<number | undefined>();
+    // tracks data changes to row currently being edited
+    // TODO: build proper loading state
+    const [updatedRow, setUpdatedRow] = useState<TableRow | null>(null);
+
+    const editableRowButtonValue = (isEditing: boolean) =>
+        isEditing ? "Save" : "Edit";
+
+    const updateRow = (value: unknown, field: string) => {
+        // largely here for typecheck reasons, this represents a case
+        // where we're somehow trying to update a row without editing enabled
```

Comment:
it is... not. It was my way of working around the fact that `editing` as a type can be `undefined`. the way the code is structured downstream in `ColumnData` there wouldn't really ever be a situation where `updateRow` would be called in a case where `editing` is undefined, but typescript doesn't know about that. It's not a very useful null check here, but in my mind it doesn't hurt us either. 

We could pass a noop to ColumnData in the case where editing is undefined, but typescript still complains without the null check. If you want, maybe we can put our heads together on a better solution.

Reply:
This looks like it's only ever called when the row is being edited, so you can override the undefined possibility with `!!`. Also, that `unknown` should be better typed, we try to avoid ambiguous types. Is a union type applicable or is there a large variance of types coming? We could use generics if so.

Reply:
talked this through, and settled on typing this value as `string` for now, with more to come as we bring this into more complex use cases